public/app/plugins/datasource/testdata/components/SimulationQueryEditor.tsx

```diff
@@ -79,6 +157,63 @@ export const SimulationQueryEditor = ({ onChange, query }: EditorProps) => {
           <Input type="text" placeholder="optional" value={simQuery.key.uid} onChange={onUIDChanged} />
         </InlineField>
       </InlineFieldRow>
+      <div>
+        <Form onSubmit={onSubmitChange}>
+          {() => (
+            <FieldSet label="Config">
+              <InlineField label="Tank capacity">
+                <Input
+                  type="number"
+                  name="tank capacity"
+                  value={cfg.tankCapacity}
+                  onChange={(e: FormEvent<HTMLInputElement>) => {
+                    setCfg({ ...cfg, tankCapacity: e.currentTarget.valueAsNumber });
```

Comment:
🤔 how does this work with different simulations?

Perhaps, lets start with the simple JSON based version, and then move to one based on forms.  BUT the forms will need to read the schema to know what to render!

Reply:
Can you update *this* PR so it takes a JSON blob in textarea... and then lets make another that tries to make a nice form -- but it will need to look at schema (part of info) to decide what fields to show